lib/app/index.js

```diff
@@ -1,53 +1,171 @@
-var async = require('async');
+'use strict';
 
-var mage = require('../mage');
-var logger = mage.core.logger;
+const memoize = require('memoizee');
 
 
 // app registration
 
-var appMap = {};
-var appList = [];
+const appMap = {};
+const appList = [];
 
 
 /**
- * This function creates instances of all configured apps and registers them into appMap and appList.
+ * Returns the configuration for all apps (overridden during unit tests)
+ *
+ * @returns {Object}
+ */
+/* istanbul ignore next */
+exports.getAppsConfig = memoize(function () {
+	return require('lib/mage').core.config.get(['apps']);
```

Comment:
I should add, I chose this memoization library for its [performance claims](https://github.com/medikoo/memoizee#benchmarks), but even then I'm sure we could optimize further (since we don't memoize things that receive arguments in these APIs). Right now, I've not prioritized this though. First we needed to establish this unit testing pattern. How we memoize is not as relevant, we can always swap that for something else.